src/Controller/SearchController.php

```diff
@@ -238,6 +251,89 @@ public function resultsAction(
+                $request = $this->setSubParamForRequestQueryParam('selectedSavedSearchTitle', null, 'search_filter', $request);
+                $searchURL = $this->getUpdatedRequestUriForRequest($request);
+
+                $redirectResponse = new RedirectResponse($request->getSchemeAndHttpHost() . $searchURL);
+
+                return $redirectResponse;
+
+            } elseif ($buttonName === 'save') {
+                // this handles cases where the "Save" button (in the "Manage my views" form part) was clicked
+                // with either "New view" or an existing saved search (aka "view") selected in the view dropdown
+
+                if (!$savedSearch) { // create a new saved search
+                    $savedSearch = new SavedSearch();
+                    $portalUserId = $currentUser->getRelatedPortalUserItem()->getItemId();
+                    $savedSearch->setAccountId($portalUserId);
+                }
+
+                $savedSearchTitle = $searchData->getSelectedSavedSearchTitle();
+                if (empty($savedSearchTitle)) {
+                    // this shouldn't get hit due to the validation annotation `@Assert\NotBlank(...)` for `SearchData->selectedSavedSearchTitle`
+                    $savedSearchTitle = $translator->trans('New view', [], 'search');
+                }
+                if ($savedSearchTitle !== $savedSearch->getTitle()) {
+                    $savedSearch->setTitle($savedSearchTitle);
+                }
+
+                // remove the "save" param from the search URL to be persisted
+                $request = $this->setSubParamForRequestQueryParam('save', null, 'search_filter', $request);
+                $savedSearchURL = $this->getUpdatedRequestUriForRequest($request);
+
+                if ($savedSearchURL !== $savedSearch->getSearchUrl()) {
+                    $savedSearch->setSearchUrl($savedSearchURL);
+                }
+
+                // for a newly created saved search, update its search URL with the correct ID
+                if (empty($savedSearch->getId())) {
+                    // persisting the new SavedSearch object will auto-assign an ID
+                    $entityManager->persist($savedSearch);
+                    $entityManager->flush();
+                    $savedSearchId = $savedSearch->getId();
+
+                    // update saved search ID in current search URL
+                    $request = $this->setSubParamForRequestQueryParam('selectedSavedSearch', $savedSearchId, 'search_filter', $request);
+                    $savedSearchURL = $this->getUpdatedRequestUriForRequest($request);
+
+                    $savedSearch->setSearchUrl($savedSearchURL);
+                }
+
+                $entityManager->persist($savedSearch);
+                $entityManager->flush();
+
+                $redirectResponse = new RedirectResponse($request->getSchemeAndHttpHost() . $savedSearchURL);
+
+                return $redirectResponse;
```

Comment:
Hier wäre redirectToRoute(...) schöner.

```suggestion
                return $this->redirect($request->getSchemeAndHttpHost() . $savedSearchURL);
```